magenta/models/music_vae/js/index.ts

```diff
@@ -198,115 +189,90 @@ class Decoder {
       return dl.stack(samples, 1) as dl.Tensor3D;
     });
   }
-
-}
-
-
-export const isDeviceSupported:boolean = isWebGLSupported() && !isSafari();
-
-
-function initialize(checkpointURL: string) {
-	const reader = new CheckpointLoader(checkpointURL);
-	return reader.getAllVariables().then(
-		(vars: { [varName: string]: dl.Tensor }) => {
-			const encLstmFw = new LayerVars(
-				vars['encoder/cell_0/bidirectional_rnn/fw/multi_rnn_cell/cell_0/lstm_cell/kernel'] as dl.Tensor2D,
-				vars['encoder/cell_0/bidirectional_rnn/fw/multi_rnn_cell/cell_0/lstm_cell/bias'] as dl.Tensor1D);
-			const encLstmBw = new LayerVars(
-				vars['encoder/cell_0/bidirectional_rnn/bw/multi_rnn_cell/cell_0/lstm_cell/kernel'] as dl.Tensor2D,
-				vars['encoder/cell_0/bidirectional_rnn/bw/multi_rnn_cell/cell_0/lstm_cell/bias'] as dl.Tensor1D);
-			const encMu = new LayerVars(
-				vars['encoder/mu/kernel'] as dl.Tensor2D,
-				vars['encoder/mu/bias'] as dl.Tensor1D);
-			const encPresig = new LayerVars(
-				vars['encoder/sigma/kernel'] as dl.Tensor2D,
-				vars['encoder/sigma/bias'] as dl.Tensor1D);
-
-			let decLstmLayers: LayerVars[] = [];
-			let l = 0;
-			while (true) {
-				const cell_prefix = DECODER_CELL_FORMAT.replace('%d', l.toString());
-				if (!(cell_prefix + 'kernel' in vars)) {
-					break;
-				}
-				decLstmLayers.push(new LayerVars(
-					vars[cell_prefix + 'kernel'] as dl.Tensor2D,
-					vars[cell_prefix + 'bias'] as dl.Tensor1D));
-				++l;
-			}
-
-			const decZtoInitState = new LayerVars(
-				vars['decoder/z_to_initial_state/kernel'] as dl.Tensor2D,
-				vars['decoder/z_to_initial_state/bias'] as dl.Tensor1D);
-			const decOutputProjection = new LayerVars(
-				vars['decoder/output_projection/kernel'] as dl.Tensor2D,
-				vars['decoder/output_projection/bias'] as dl.Tensor1D);
-			let nade = (('decoder/nade/w_enc' in vars) ?
-				new Nade(
-					vars['decoder/nade/w_enc'] as dl.Tensor3D,
-					vars['decoder/nade/w_dec_t'] as dl.Tensor3D) : null);
-			return [
-				new Encoder(encLstmFw, encLstmBw, encMu, encPresig),
-				new Decoder(decLstmLayers, decZtoInitState, decOutputProjection, nade)
-			];
-		})
 }
 
 class MusicVAE {
-
-	checkpointURL:string;
+  checkpointURL:string;
   encoder: Encoder;
   decoder: Decoder;
+  rawVars: {[varName: string]: dl.NDArray<dl.Rank>};
```

Comment:
Two comments: 
1) no need for dl.RANK
2) Use `dl.Tensor`

Reply:
Done